modules/nextflow/src/main/groovy/nextflow/data/cid/CidStore.groovy

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2013-2024, Seqera Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package nextflow.data.cid
+
+import java.nio.file.Path
+import java.util.function.Consumer
+
+import groovy.transform.CompileStatic
+import nextflow.data.config.DataConfig
+/**
+ *
+ * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
+ */
+@CompileStatic
+interface CidStore {
+
+    void open(DataConfig config)
+
+    void save(String key, Object value)
+
+    void list(String key, Consumer<String> consumer)
```

Comment:
I was writing the JavaDoc for the interfaces and I have seen this function is not used.  @pditommaso, do you remember what was the purpose of this method? In the DefaultCidStore it is returning the files in the key, but currently it is just the .data.json file, because data is not stored in the CidStore. Not sure if it is needed anymore.

Reply:
Yes the list. I haven't modified it. It was as it is from the first commit. But it has never been used. So, I will remove it. I will add a similar one in case it is needed in the future.